class-job.php

```diff
@@ -165,12 +166,17 @@ public static function get_by_site( $site, $include_completed = false ) {
 			return new WP_Error( 'cavalcade.job.invalid_site_id' );
 		}
 
+		$statuses = array( 'waiting', 'running' );
+		if ( $include_completed ) {
+			$statuses[] = 'completed';
+		}
+		if ( $include_failed ) {
+			$statuses[] = 'failed';
+		}
+
 		// Find all scheduled events for this site
 		$table = static::get_table();
-		$sql = "SELECT * FROM `{$table}` WHERE site = %d";
-		if ( ! $include_completed ) {
-			$sql .= ' AND status != "completed"';
-		}
+		$sql = "SELECT * FROM `{$table}` WHERE site = %d AND status IN('" . implode( "','", $statuses ) . "')";
```

Comment:
I'd prefer this to use placeholders, i.e. `implode( ',', array_fill( 0, count( $statuses ), '%s' ) )`